apple/internal/templates/apple_simulator.template.py

```diff
@@ -444,37 +454,71 @@ def temporary_simulator(
               prefix="com.apple.CoreSimulator.SimRuntime",
               runtime_platform=runtime_platform,
               runtime_version_name=runtime_version_name,
-          )
+          ),
       ],
       encoding="utf-8",
       check=True,
-      stdout=subprocess.PIPE)
+      stdout=subprocess.PIPE,
+  )
   udid = simctl_create_result.stdout.rstrip()
   try:
     logger.info("Killing all running simulators...")
-    subprocess.run(["pkill", "Simulator"],
-                   stderr=subprocess.DEVNULL,
-                   check=False)
+    subprocess.run(
+        ["pkill", "Simulator"], stderr=subprocess.DEVNULL, check=False
+    )
     yield udid
   finally:
     logger.info("Shutting down simulator with udid: %s", udid)
-    subprocess.run([simctl_path, "shutdown", udid],
-                   stderr=subprocess.DEVNULL,
-                   check=False)
+    subprocess.run(
+        [simctl_path, "shutdown", udid], stderr=subprocess.DEVNULL, check=False
+    )
     logger.info("Deleting simulator with udid: %s", udid)
     subprocess.run([simctl_path, "delete", udid], check=True)
 
 
+def register_dsyms(dsyms_dir: str):
+  """Adds all dSYMs in `dsyms_dir` to the symbolscache.
+
+  Args:
+    dsyms_dir: Path to directory potentially containing dSYMs
+  """
+  symbolscache_command = [
+      "/usr/bin/symbolscache",
+      "delete",
+      "--tag",
+      "Bazel",
+      "compact",
+      "add",
+      "--tag",
+      "Bazel",
+  ] + [
+      a
+      for a in pathlib.Path(dsyms_dir).glob(
```

Comment:
Is this glob ever going to be empty?

Reply:
yea if your build command doesn't generate dsyms i think it will be

Reply:
when it's empty symbolscache just deletes in that case:

```
2023-08-25 11:36:31.979 INFO Running command: ['/usr/bin/symbolscache', 'delete', '--tag', 'Bazel', 'compact', 'add', '--tag', 'Bazel']
2023-08-25 11:36:32.119 INFO symbolscache output:   No entries deleted
```